pkg/controllers/user/helm/controller.go

```diff
@@ -150,8 +150,13 @@ func (l *Lifecycle) Updated(obj *v3.App) (runtime.Object, error) {
 				},
 			}
 			_, err := l.NsClient.Create(&n)
-			if err != nil && !errors.IsAlreadyExists(err) {
-				return obj, err
+			if err != nil {
+				if !errors.IsAlreadyExists(err) {
+					return obj, err
+				}
+				if n.DeletionTimestamp != nil {
```

Comment:
are we checking the deletion timestamp on the correct object? Seems like we are doing it on the namespace-to-create

Reply:
not sure how I missed that, thank you!!